source/client/stream_decoder.h

```diff
@@ -81,6 +81,10 @@ class StreamDecoder : public Envoy::Http::StreamDecoder,
   streamResetReasonToResponseFlag(Envoy::Http::StreamResetReason reset_reason);
   void finalizeActiveSpan();
   void setupForTracing(std::string& x_request_id);
+  static const std::string& staticUploadContent() {
+    static const std::string s(4194304, 'a');
```

Comment:
This needs to be a `new` and a pointer as per `CONSTRUCT_ON_FIRST_USE`, otherwise you have static destruction fiasco.

Reply:
I read up over at https://isocpp.org/wiki/faq/ctors#construct-on-first-use-v2 and followed the approach for a when a non-trivial destructor is involved -- however, reading more, it's not like the `std::string` is needs to write to a file or some such in its destructor, so I'll amend. Sorry about the noise.

Reply:
Amended